com.unity.render-pipelines.high-definition/Runtime/Lighting/ScreenSpaceLighting/BilateralUpsample.hlsl

```diff
@@ -38,7 +38,46 @@ float4 BilUpColor(float HiDepth, float4 LowDepths, float4 lowValue0, float4 lowV
     return WeightedSum / TotalWeight;
 }
 
-static const float BilateralUpSampleWeights3x3[9] = {9.0, 3.0, 3.0, 3.0, 3.0, 1.0, 1.0, 1.0, 1.0};
+// The bilateral upscale function (2x2 neighborhood) (single channel version)
+float BilUpSingle(float HiDepth, float4 LowDepths, float4 lowValue)
+{
+    float4 weights = float4(9, 3, 1, 3) / (abs(HiDepth - LowDepths) + _UpsampleTolerance);
+    float TotalWeight = dot(weights, 1) + _NoiseFilterStrength;
+    float WeightedSum = dot(lowValue, weights) + _NoiseFilterStrength;
+    return WeightedSum / TotalWeight;
+}
+
+// This is the representation of the halfresolution neighborhood
+// |-----|-----|-----|
+// |     |     |     |
+// |-----|-----|-----|
+// |     |     |     |
+// |-----|-----|-----|
+// |     |     |     |
+// |-----|-----|-----|
+
+// This is the representation of the fullresolution neighborhood
+// |-----|-----|-----|
+// |     |     |     |
+// |-----|--|--|-----|
+// |     |--|--|     |
+// |-----|--|--|-----|
+// |     |     |     |
+// |-----|-----|-----|
+
+// The base is centered at (0, 0) at the center of the center pixel:
+// The 4 full res pixels are centered {L->R, T->B} at {-0.25, -0.25}, {0.25, -0.25}
+//                                                    {-0.25, 0.25}, {0.25, 0.25}
+//
+// The 9 half res pixels are placed {L->R, T->B} at {-1.0, -1.0}, {0.0, -1.0}, {1.0, -1.0}
+//                                                  {-1.0, 0.0}, {0.0, 0.0}, {1.0, 0.0}
+//                                                  {-1.0, 1.0}, {0.0, 1.0}, {1.0, 1.0}
+
+// Set of pre-generated weights (L->R, T->B). After experimentation, the final weighting function is exp(-distance^2)
+static const float distanceBasedWeights[9 * 4] = {0.324652, 0.535261, 0.119433, 0.535261, 0.882497, 0.196912, 0.119433, 0.196912, 0.0439369,
```

Comment:
This kind of code gets at times compiled into an horrible mess, I suggest moving this to a constant buffer (looks like it'll be correctly aligned already so should be straightforward)